File: boj/1260.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

    static ArrayList<Integer>[] arr = new ArrayList[1003];
    static boolean[] check;
    static StringBuilder sb = new StringBuilder();
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int v = Integer.parseInt(st.nextToken());
        for(int i = 1; i <= n; ++i) arr[i] = new ArrayList<>();
        for(int i = 0; i < m; ++i) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            arr[a].add(b); arr[b].add(a);
        }
        for(int i = 1; i <= n; ++i) Collections.sort(arr[i]);
        check = new boolean[n + 1];
        dfs(v);
        sb.append("\n");
        check = new boolean[n + 1];
        bfs(v);
        System.out.println(sb);
    }

    public static void dfs(int num) {
        check[num] = true;
        sb.append(num).append(" ");
        int len = arr[num].size();
        for(int i = 0; i < len; ++i) {
            if(!check[arr[num].get(i)]) dfs(arr[num].get(i));
        }
    }

    public static void bfs(int num) {
        Queue<Integer> q = new LinkedList<Integer>();
        q.add(num);
        check[num] = true;
        while(!q.isEmpty()) {
            int temp = q.remove();
            sb.append(temp).append(" ");
            int len = arr[temp].size();
            for(int i = 0; i < len; ++i) {
                if(!check[arr[temp].get(i)]) {
                    q.add(arr[temp].get(i));
                    check[arr[temp].get(i)] = true;
                }
            }
        }
        sb.append("\n");
    }
}

File: boj/1260.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> arr[1003];
int check[1003];
queue<int> q;

void dfs(int num) {
    check[num] = 1;
    cout << num << " ";
    int len = arr[num].size();
    for(int i = 0; i < len; ++i) {
        if(!check[arr[num][i]]) dfs(arr[num][i]);
    }
}

void bfs(int num) {
    q.push(num);
    check[num] = 1;
    while(!q.empty()) {
        int temp = q.front();
        cout << temp << " ";
        q.pop();
        int len = arr[temp].size();
        for(int i = 0; i < len; ++i) {
            if(!check[arr[temp][i]]) {
                q.push(arr[temp][i]);
                check[arr[temp][i]] = 1;
            }
        }
    }
    cout << "\n";
}

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    int n, m, v; cin >> n >> m >> v;
    for(int i = 0; i < m; ++i) {
        int a, b; cin >> a >> b;
        arr[a].push_back(b); arr[b].push_back(a);
    }
    for(int i = 1; i <= n; ++i) sort(arr[i].begin(), arr[i].end());
    dfs(v);
    cout << "\n";
    for(int i = 0; i <= n; ++i) check[i] = 0;
    bfs(v);
    return 0;
}